server/paymentProviders/opencollective/host.js

```diff
@@ -12,6 +15,26 @@ paymentMethodProvider.features = {
   waitToCharge: false,
 };
 
+paymentMethodProvider.refundTransaction = async (transaction, user) => {
+  if (transaction.CollectiveId === transaction.FromCollectiveId) {
+    throw new Error('Cannot refund a transaction from the same collective');
+  }
+
+  const host = await models.Collective.findByPk(transaction.CollectiveId);
+
+  const balance = await host.getBalanceWithBlockedFunds();
+  if (balance < transaction.amount) {
```

Comment:
How do you ensure there is no currency mismatch?

Reply:
Ops, I've corrected this by passing the `transaction.currency` when obtaining the balance amount.